data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -139,22 +151,36 @@ private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retr
       cause -> {
         if (cause instanceof ResponseFailureException) {
           final var response = ((ResponseFailureException) cause).getResponse();
+
+          Tags tags;
+          if (response == null) {
+            tags = this.noResponseResourceTags;
+          } else {
+            tags = this.consumerVerticleContext.getTags().and(
+              Tag.of(Metrics.Tags.RESPONSE_CODE_CLASS, response.statusCode() / 100 + "xx"),
+              Tag.of(Metrics.Tags.RESPONSE_CODE, Integer.toString(response.statusCode())));
+          }
```

Comment:
We only need tags when the `if` at line 164 is true, so we should do this in that block